src/engine/engine.js

```diff
@@ -35,21 +35,44 @@ function createState(initState) {
   };
 }
 
+class ContextWrapper extends Component {
+  getChildContext() {
+    return { store: this.store }
+  }
+
+  constructor(props, context) {
+    super(props, context);
+    this.store = props.store
+  }
+
+  render() {
+    return Children.only(this.props.children)
+  }
+}
+
+ContextWrapper.childContextTypes = {
+  store: React.PropTypes.any
+};
+
 export default class Engine extends Component {
   constructor(props, context) {
     super(props, context);
     this.state = createState(props.initState);
+
+    this.updateComponentState = this.updateComponentState.bind(this);
+
+    sagaMiddleware.run(watchReadPerform);
+    this.state.store.subscribe(this.updateComponentState);
   }
 
   updateComponentState() {
```

Comment:
This method should be removed. Instead of, on subscription we should trigger this.forceUpdate(), so that triggers / propagates the changes.